carmin-server/server/resources/path.py

```diff
@@ -30,47 +35,66 @@ def get(self, complete_path: str = ''):
-
-        if not os.path.exists(requested_data_path):
-            return ErrorCodeAndMessageMarshaller(INVALID_PATH)
+        if not is_safe_path(full_absolute_path):
+            return ErrorCodeAndMessageMarshaller(UNAUTHORIZED), 403
+        if not os.path.exists(full_absolute_path) and action != 'exists':
+            return ErrorCodeAndMessageMarshaller(INVALID_PATH), 401
 
         if action == 'content':
-            return content_action(requested_data_path)
+            return get_content(full_absolute_path)
         elif action == 'properties':
-            path = properties_action(data_path, complete_path)
+            path = PathModel.object_from_pathname(full_absolute_path)
             return PathSchema().dump(path)
         elif action == 'exists':
-            pass
+            path_exists = os.path.exists(full_absolute_path)
+            return BooleanResponseSchema().dump(BooleanResponse(path_exists))
         elif action == 'list':
-            if not os.path.isdir(requested_data_path):
+            if not os.path.isdir(full_absolute_path):
                 return ErrorCodeAndMessageMarshaller(LIST_ACTION_ON_FILE)
-            directory_list = list_action(data_path, complete_path)
+            directory_list = get_list(data_path, complete_path)
             return PathSchema(many=True).dump(directory_list)
         elif action == 'md5':
-            if os.path.isdir(requested_data_path):
-                return ErrorCodeAndMessageMarshaller(MD5_ON_DIR)
-            md5 = PathMD5(generate_md5(requested_data_path))
+            if os.path.isdir(full_absolute_path):
+                return ErrorCodeAndMessageMarshaller(MD5_ON_DIR), 400
+            md5 = generate_md5(full_absolute_path)
             return PathMD5Schema().dump(md5)
         else:
-            return ErrorCodeAndMessageMarshaller(INVALID_ACTION)
+            return ErrorCodeAndMessageMarshaller(INVALID_ACTION), 400
 
-    def put(self, complete_path):
-        pass
+    def put(self, complete_path: str = ''):
+        data_path = app.config['DATA_DIRECTORY']
+        requested_data_path = os.path.normpath(
+            os.path.join(data_path, complete_path))
+
+        if not is_safe_path(data_path, requested_data_path):
+            return ErrorCodeAndMessageMarshaller(UNAUTHORIZED), 403
+        if not parent_dir_exists(requested_data_path):
+            return ErrorCodeAndMessageMarshaller(INVALID_PATH), 401
+        upload_data = request.get_json(force=True, silent=True)
+
+        if not upload_data:
+            return create_directory(requested_data_path)
+        elif upload_data["type"] == "File":
+            upload_data = UploadDataSchema().load(upload_data).data
+            return upload_file(upload_data, requested_data_path)
+        elif upload_data["type"] == "Archive":
+            upload_data = UploadDataSchema().load(upload_data).data
+            return upload_dir(upload_data, requested_data_path)
```

Comment:
For clarity purposes, I would rename upload_archive in case we want to actually have a way to upload to a dir later (specifying an unexistant dir and file in base64)